[Feature][Connector] support avro format

liunaijie opened the pull request:
<!--

Thank you for contributing to SeaTunnel! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/seatunnel/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

-->

## Purpose of this pull request
close #4821 

<!-- Describe the purpose of this pull request. For example: This pull request adds checkstyle plugin.-->

- added avro format and UT
- kafka support avro format and e2e test

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

liunaijie committed:
[Fea] support avro format

- added avro format and UT
- kafka support avro format and e2e test

liunaijie committed:
[Feature] fix code style and update avro version

Signed-off-by: Jarvis <liunaijie1996@163.com>

TyrantLucifer commented:
Good pr, let's waiting CI/CD

liunaijie commented:
pls re-trigger the CI.

liugddx commented:
Can you improve the relevant documents?

https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/source/kafka.md
https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/sink/kafka.md

wssmao committed:
[Bug] [zeta][starter]fix bug (#4983) (#4984)

Co-authored-by: wsstony <tonymao777@163.com>

lightzhao committed:
fix redis nodes format error. (#4981)

Co-authored-by: lightzhao <zhaolianyong777@gmail.com>

liugddx committed:
[Improve][CDC]Remove  driver for cdc connector (#4952)

MonsterChenzhuo committed:
[Hotfix][Connector-V2][Mongodb] Fix document error content and remove redundant code (#4982)

Co-authored-by: chenzy15 <chenzy15@ziroom.com>

TyrantLucifer committed:
[Improve][Connector-V2][OSS-Jindo] Optimize jindo oss connector (#4964)

* [Improve][Connector-V2][Jindo-Oss] Optimize jindo-oss connector

* [Improve][Connector-V2][Jindo-Oss] Update module name

* [Hotfix][Connector-V2][StarRocks] Fix code style

XiaoJiang521 committed:
[bugfix] Upgrade the key log output level(#4993)

FuYouJ committed:
[Feature][Zeta] Configuration files support user variable replacement (#4969)

rewerma committed:
[Feature][Transform-V2][SQL] Support 'select *' and 'like' clause for SQL Transform plugin (#4991)

Co-authored-by: mcy <rewrma@163.com>

liunaijie commented:
@TyrantLucifer @liugddx PTAL

liugddx committed:
[Improve][CDC]change driver scope to provider (#5002)

TyrantLucifer committed:
[Hotfix][Connector-V2][Hive] Support user-defined hive-site.xml (#4965)

MonsterChenzhuo committed:
[Improve][Connector-v2][Mongodb]Optimize reading logic (#5001)

Co-authored-by: chenqqq11 <chenzy15@ziroom.com>

TaoZex commented on seatunnel-formats/seatunnel-format-avro/src/main/java/org/apache/seatunnel/format/avro/RowToAvroConverter.java:
```diff
@@ -0,0 +1,234 @@
+            default:
+                String errorMsg =
+                        String.format(
+                                "SeaTunnel file connector is not supported for this data type [%s]",
```
```suggestion
                                "SeaTunnel avro format is not supported for this data type [%s]",
```

TaoZex reviewed (approved):
LGTM

Carl-Zhou-CN committed:
[Feature][Connector-V2][Clickhouse] clickhouse writes with checkpoints (#4999)

MonsterChenzhuo committed:
[Hotfix][Connector-V2][Mongodb] Compatible with historical parameters (#4997)

liugddx commented on seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/resources/avro/kafka_avro_to_console.conf:
Is this for testing avro format with kafka as the source?  The source is not configured with avro?

liugddx commented on seatunnel-formats/seatunnel-format-avro/pom.xml:
```diff
@@ -0,0 +1,45 @@
+    <name>SeaTunnel : Formats : Avro</name>
+
+    <properties>
+        <avro.version>1.10.2</avro.version>
```
How is the dependent version defined? Why use this version? If I use hudi sink, can this format be used?

liunaijie replied:
change to 1.9.0. 
since 1.9.0 use java LocalDate and Instant as date, timestamp type. before this version avro used joda time related api.

this format complete the convert between SeaTunnelRow and Avro GenericRecord (see RowToAvroConverter and AvroToRowConverter). so in general it can be used of any source and sink.

EricJoy2048 committed:
Split updated modules integration test for part 4 (#5028)

EricJoy2048 committed:
[Hotfix] Fix the CI Job name error (#5032)

liunaijie commented:
can we re-trigger the tests, i check the error, it's failed on PaimonIT. get NPE when write data to paimon. but the col name and value all are not null. 
![image](https://github.com/apache/seatunnel/assets/28497357/84c8ac93-939a-4a8e-a0e5-97ea6d5990f4)


![image](https://github.com/apache/seatunnel/assets/28497357/78229cc5-f01d-43fc-a2b4-d39e17edc3d4)

liugddx commented:
> can we re-trigger the tests, i check the error, it's failed on PaimonIT. get NPE when write data to paimon. but the col name and value all are not null.

This looks like a bug

hailin0 committed:
[Feature][CDC] Support disable/enable exactly once for INITIAL (#4921)

ic4y committed:
[bugfix][zeta] Fixed multi-table job data loss and latency issues (#149) (#5031)

hailin0 committed:
[Hotfix][CDC] Fix jdbc connection leak for mysql (#5037)

ic4y committed:
[Bugfix][zeta] Fix cdc connection does not close (#4922)

EricJoy2048 committed:
Fix XA Transaction bug (#5020)

liunaijie commented:
I remember when use avro 1.10.2 it can pass the actions. so i will increase the avro version to verify is this related.

liunaijie commented:
increase the avro version same with paimon

yctanGmail committed:
Set Up with Kubernetes, dockerfile document error in constructing docker image (#5022)

Co-authored-by: yctan <1417983443@qq.com>

liunaijie commented:
updated to 1.10.2. now i think it can pass the CI/CD

liunaijie commented:
It's weird, in this [action](https://github.com/apache/seatunnel/actions/runs/5484872607/jobs/9993746562) I updated avro version to 1.10.2, but forgot to change the license.txt. this action pass all tests, exclude dependency check.

![image](https://github.com/apache/seatunnel/assets/28497357/4dad340b-c2be-41a1-9d67-f22ac402da66)

and when i updated the licences.txt and ru-run. fail on other stage.
![image](https://github.com/apache/seatunnel/assets/28497357/456960ab-4fca-4568-9901-a7c0778e0516)

liugddx commented:
> It's weird, in this [action](https://github.com/apache/seatunnel/actions/runs/5484872607/jobs/9993746562) I updated avro version to 1.10.2, but forgot to change the license.txt. this action pass all tests, exclude dependency check.
> 
> ![image](https://user-images.githubusercontent.com/28497357/252253399-4dad340b-c2be-41a1-9d67-f22ac402da66.png)
> 
> and when i updated the licences.txt and ru-run. fail on other stage. ![image](https://user-images.githubusercontent.com/28497357/252254066-456960ab-4fca-4568-9901-a7c0778e0516.png)

Can you see why?

MonsterChenzhuo committed:
[Improve][Connector-v2][Mongodb]sink support transaction update/writing (#5034)

wu-a-ge committed:
fix:the HdfsStorage can not delete checkpoint file #5046 (#5054)

liunaijie commented:
i run the actions on myself repo. it run successful without any change

https://github.com/liunaijie/seatunnel/actions/runs/5507761324
https://github.com/liunaijie/seatunnel/actions/runs/5507761323
https://github.com/liunaijie/seatunnel/actions/runs/5507761322

here is the code campare, you can see i don't have any change exclude the action config.
https://github.com/liunaijie/seatunnel/compare/dev...liunaijie:seatunnel:avro-1.10.2

pls re-trigger the ci, let's see the result.

BruceMaa committed:
[BugFix] [Connector-V2] [MySQL-CDC] serverId from int to long (#5033) (#5035)

* [bugfix] change MySQL CDC serverId from int to long (#5033)

* style: 🎨 optimize code style

Carl-Zhou-CN committed:
[Feature][Connector-V2][cdc] Change the time zone to the default time zone (#5030)

happyboy1024 committed:
[Bugfix][connector-cdc-mysql] Fix listener not released when BinlogClient reuse (#5011)

whhe committed:
[Feature][Connector-V2][Jdbc] Add oceanbase dialect factory (#4989)


---------

Co-authored-by: silenceland <silenceland23@163.com>
Co-authored-by: changhuyan <877018069@qq.com>

wu-a-ge committed:
[HotFix][Zeta] fix after the savepoint job is restored, the checkpoint file cannot be generated #4985 (#5051)

* fix after the savepoint job is restored, the checkpoint file cannot be generated

liugddx committed:
fix class not found exception (#5063)

liunaijie committed:
[Feature] update action config to support run CI on fork repo (#5065)
